lib/grp.gi

```diff
@@ -4491,6 +4500,23 @@ InstallMethod( MaximalNormalSubgroups, "for simple groups",
               function(G) return [ TrivialSubgroup(G) ]; end);
 
 
+#############################################################################
+##
+#M  MaximalNormalSubgroups( <G> )
+##
+InstallMethod( MaximalNormalSubgroups, "general method selection",
+              [ IsGroup ],
+    function(G)
+
+    if 0 in AbelianInvariants(G) then
+      # (p) is a maximal normal subgroup in Z for every prime p
+      return fail;
```

Comment:
Regarding this line, @markuspf previously asked whether returning `fail` here makes sense or not. And @hungaborhorvath stated that he initially had an `Error()` here, but found `fail` to be more useful. I mostly agree with the latter. And regarding the possibility of returning an infinite iterator here: Sure, that might be possible for some cases at least, but I also doubt it'd be that useful... Perhaps some day somebody will arrive at a situation where it *is* useful, and then we can revisit this. Until then, I think it would be a waste of time to implement such an iterator, plus, let's not forgot the golden rule "code you don't actually use is always wrong" ;-).

We could of course add a new constant `UnableToCompute` or `ResultIsInfinite` or so, but again: as long as nobody actually *uses* this...

Finally, the "rule" to a degree even applies to `fail`. @hungaborhorvath do you use this, or is this just a case of "might be useful"? In the latter case, I'd just put an error here. Then, if we later find a use case where we should return fail and/or some other special constant, we can reopen this, without breaking any existing code. While if we return `fail` now, we can never again change it without risking to break somebody's code... So in this sense, `Error` is the safe and future proof solution :-)